crate/server/src/core/implementation.rs

```diff
@@ -1,50 +1,64 @@
+    pub(crate) params: ServerParams,
+    pub(crate) db: Box<dyn Database + Sync + Send>,
+    // pub(crate) findex_configuration: Configuration,
+}
 
 impl FindexServer {
     pub(crate) async fn instantiate(mut shared_config: ServerParams) -> FResult<Self> {
-        let db: Box<dyn Database + Sync + Send> = if let Some(mut db_params) =
-            shared_config.db_params.as_mut()
-        {
-            match &mut db_params {
-                DbParams::Sqlite(db_path) => Box::new(
-                    SqlitePool::instantiate(
-                        &db_path.join("findex_server.db"),
-                        shared_config.clear_db_on_start,
-                    )
-                    .await?,
-                ),
-                DbParams::RedisFindex(url, master_key, label) => {
-                    // There is no reason to keep a copy of the key in the shared config
-                    // So we are going to create a "zeroizable" copy which will be passed to Redis with Findex
-                    // and zeroize the one in the shared config
-                    let new_master_key =
-                        Secret::<REDIS_WITH_FINDEX_MASTER_KEY_LENGTH>::from_unprotected_bytes(
-                            &mut master_key.to_bytes(),
-                        );
-                    // `master_key` implements ZeroizeOnDrop so there is no need
-                    // to manually zeroize.
-                    Box::new(
-                        RedisWithFindex::instantiate(url.as_str(), new_master_key, label).await?,
-                    )
+        let db: Box<dyn Database + Sync + Send> =
+            if let Some(mut db_params) = shared_config.db_params.as_mut() {
+                match &mut db_params {
+                    DbParams::Redis(url) => Box::new(Redis::instantiate(url.as_str()).await?),
                 }
-            }
-        } else {
-            findex_server_bail!("Fatal: no database configuration provided. Stopping.")
-        };
+            } else {
+                findex_server_bail!("Fatal: no database configuration provided. Stopping.")
+            };
 
         Ok(Self {
             params: shared_config,
             db,
         })
     }
+
+    /// Get the user from the request depending on the authentication method
+    /// The user is encoded in the JWT `Authorization` header
+    /// If the header is not present, the user is extracted from the client
+    /// certificate If the client certificate is not present, the user is
+    /// extracted from the configuration file
+    pub(crate) fn get_user(&self, req_http: &HttpRequest) -> String {
+        let default_username = self.params.default_username.clone();
+
+        if self.params.force_default_username {
+            debug!(
+                "Authenticated using forced default user: {}",
+                default_username
+            );
+            return default_username;
+        }
+        // if there is a JWT token, use it in priority
```

Comment:
I am not sure we want to support username+password.

Reply:
It is not a username+password authentication but rather a JWT or PKCS12 mutual authentication